torch/testing/_internal/custom_op_db.py

```diff
@@ -275,6 +380,19 @@ def numpy_split_copy_with_int_backward(ctx, grad_out, _):
     numpy_split_copy_with_int_backward,
     setup_context=numpy_split_copy_with_int_setup_context)
 
+def numpy_split_copy_with_int_vmap(info, in_dims, x, splits, dim):
+    x_bdim, _ , _ = in_dims
+    x = x.movedim(x_bdim, 0)
+    result, len_split = numpy_split_copy_with_int(x, splits, dim + 1)
+    return (result, len_split), ([0 for _ in range(len(result))], None)
+
+numpy_split_copy_with_int.register_vmap(numpy_split_copy_with_int_vmap)
+
+def sample_vmap_out_dim_numpy_split_copy_with_int(x, splits, dim):
+    return [0 for _ in range(len(splits) + 1)], None
+
+add_vmap_out_dims_example_custom_op('NumpySplitCopyWithIntCustomOp', sample_vmap_out_dim_numpy_split_copy_with_int)
```

Comment:
Sorry, I answered your question over chat out of context. Many operators that return int cannot actually be vmapped, this is one example. In this case it would be good if we prevented the user from adding a vmap rule (and if we automatically added one that errors out)

The reason why vmap cannot support an int output is because it should actually return multiple ints, but the schema of the operator says it returns a single int. The solution here for a user is to change their int return to a Tensor return. Since we're just in testing code here we should error out.

Reply:
I have a similar question as above, in the default case, if we let the `out_dim` be None, then it only returns a single int as the schema suggests?

```python
def f(x):
    return x, 1

x = torch.randn(3)
print(torch.vmap(f, out_dims=(0, None))(x))
# (tensor([-0.6958, -0.3569, -0.3476]), 1)
```

Reply:
Resolved